### ISSUE TITLE
Build fails with clang due to incorrect return types

### ISSUE THREAD

**Issue by ilovezfs** [elboza/dwarf-ng#2]
Granted, this code is unreachable, but it causes the build to fail with clang unless return-type warnings/errors are disabled, which wouldn't be particularly safe to do.

```
libdwarf.c:338:43: error: non-void function 'growth' should return a value [-Wreturn-type]
        if(!fc_ptr) {printf("no file opened!\n");return;}
```

``` ^
libdwarf.c:360:43: error: non-void function 'shrink' should return a value [-Wreturn-type]
        if(!fc_ptr) {printf("no file opened!\n");return;}
```

```
dw_switchers.c:158:42: error: void function 'set_s_val' should not return a value [-Wreturn-type]
        if(!fc_ptr) {printf("unknown file.\n"); return NULL;}
```

```
dw_pe_utils.c:575:14: error: non-void function 'get_pe_offset2' should return a value [-Wreturn-type]
        if(!fc_ptr) return;
```

Patch:

```
diff --git a/src/libdwarf/dw_pe_utils.c b/src/libdwarf/dw_pe_utils.c
index f4245f0..7564961 100644
--- a/src/libdwarf/dw_pe_utils.c
+++ b/src/libdwarf/dw_pe_utils.c
@@ -572,7 +572,7 @@ off_t get_pe_offset2(int sect,int num,int endoffset)
        off_t offs;
        char ss[15];

-       if(!fc_ptr) return;
+       if(!fc_ptr) return 0;
        maxsect=get_max_pesect();
        mz=(_IMAGE_DOS_HEADER*)fc_ptr->faddr;
        peoff=get_data32(mz->e_lfanew);
diff --git a/src/libdwarf/dw_switchers.c b/src/libdwarf/dw_switchers.c
index c45c72f..fac007a 100644
--- a/src/libdwarf/dw_switchers.c
+++ b/src/libdwarf/dw_switchers.c
@@ -155,7 +155,7 @@ struct _var* get_s_val(struct _structvar *ptr)
 }
 void set_s_val(struct _structvar *ptr,struct _var *val)
 {
-       if(!fc_ptr) {printf("unknown file.\n"); return NULL;}
+       if(!fc_ptr) {printf("unknown file.\n"); return;}
        switch(fc_ptr->file_type){
                case FT_ELF:
                        elf_set_s_val(ptr,val);
diff --git a/src/libdwarf/libdwarf.c b/src/libdwarf/libdwarf.c
index 2bff251..4fd418a 100644
--- a/src/libdwarf/libdwarf.c
+++ b/src/libdwarf/libdwarf.c
@@ -335,7 +335,7 @@ int growth(off_t len)
        off_t offset;
        int n;
        char *x;
-       if(!fc_ptr) {printf("no file opened!\n");return;}
+       if(!fc_ptr) {printf("no file opened!\n");return false;}
        if(!fc_ptr->can_grow) {printf("this file cannot change its size\n"); return false;}
        if(!fc_ptr->fd) {printf("no file opened!\n");return false;}
        x=(char*)malloc(len);
@@ -357,7 +357,7 @@ int shrink(off_t len)
 {
        off_t offset,new_offset;
        int n;
-       if(!fc_ptr) {printf("no file opened!\n");return;}
+       if(!fc_ptr) {printf("no file opened!\n");return false;}
        if(!fc_ptr->can_grow) {printf("this file cannot change its size\n"); return false;}
        if(!fc_ptr->fd) {warn("no file opened!\n");return false;}
        offset=lseek(fc_ptr->fd,(off_t)0,SEEK_END);

```

**Comment by ilovezfs** [comment]
@elboza ping on this.

**Comment by ilovezfs** [comment]
@elboza any update here? dwarf-ng was boneyarded in Homebrew because this remains unfixed.

**Comment by elboza** [comment]
solved with dwarf-0.3.2

**Comment by ilovezfs** [comment]
Thanks @elboza!
